src/OmniSharp.Cake/CakeProjectSystem.cs

```diff
@@ -148,7 +148,7 @@ private void ScriptReferencesChanged(object sender, ReferencesChangedEventArgs e
                 var document = solution.GetDocument(documentId);
                 var project = document.Project;
 
-                var metadataReferences = e.References.Select(reference => MetadataReference.CreateFromFile(reference, documentation: GetDocumentationProvider(reference)));
+                var metadataReferences = GetMetadataReferences(e.References);
```

Comment:
you may want to use https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Roslyn/Utilities/MetadataReferenceEqualityComparer.cs to avoid duplicates

Reply:
Actually, `e.References` should be `ISet<string>` and not `IReadOnlyCollection<string>`, so duplicates *should* already be avoided.

Reply:
@filipw is referring to the issue of adding the same assembly reference to a project twice. This can happen if there are multiple copies of the same assembly with different file paths.